src/tests/functional/plugin/cpu/single_layer_tests/softmax.cpp

```diff
@@ -1,222 +1,227 @@
+
+        auto inputShape = targetStaticShapes.front().front();
+
+        auto params = ngraph::builder::makeDynamicParams(inType, inputDynamicShapes);
+
+        const auto paramOuts =
+            ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
+
+        const auto softMax = std::make_shared<ngraph::opset1::Softmax>(paramOuts.at(0), config.axis);
+
+        function = makeNgraphFunction(inType, params, softMax, "SoftMax");
+    }
+};
+
+TEST_P(SoftMaxLayerCPUTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+    run();
+    CheckPluginRelatedResults(executableNetwork, "Softmax");
+}
+
+namespace {
+// not optimized cpu spec
+const auto notOptimizedCPUSpec = CPUSpecificParams{{}, {}, {"ref_any"}, "ref_any"};
+
+const std::vector<SoftMaxConfig> optimizedConfigsFP32 = {
+    // Static shapes
+    {ShapesDefenition{{ov::PartialShape{1, 100}}, {{ov::Shape{1, 100}, ov::Shape{1, 100}}}}, 1},
+    {ShapesDefenition{{ov::PartialShape{10, 10}}, {{ov::Shape{10, 10}}}}, 1},
+    {ShapesDefenition{{ov::PartialShape{100, 1}}, {{ov::Shape{100, 1}}}}, 0},
+    {ShapesDefenition{{ov::PartialShape{100, 1}}, {{ov::Shape{100, 1}}}}, 1},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 1}}, {{ov::Shape{5, 5, 1}}}}, 1},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 5}}, {{ov::Shape{5, 5, 5}}}}, 2},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 5, 5}}, {{ov::Shape{5, 5, 5, 5}}}}, 0},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 1, 1}}, {{ov::Shape{5, 5, 1, 1}}}}, 1},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 5, 5}}, {{ov::Shape{5, 5, 5, 5}}}}, 1},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 5, 1}}, {{ov::Shape{5, 5, 5, 1}}}}, 2},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 5, 5}}, {{ov::Shape{5, 5, 5, 5}}}}, 2},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 5, 5}}, {{ov::Shape{5, 5, 5, 5}}}}, 3},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 5, 5, 5}}, {{ov::Shape{5, 5, 5, 5, 5}}}}, 0},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 1, 1, 1}}, {{ov::Shape{5, 5, 1, 1, 1}}}}, 1},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 5, 5, 5}}, {{ov::Shape{5, 5, 5, 5, 5}}}}, 1},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 5, 1, 1}}, {{ov::Shape{5, 5, 5, 1, 1}}}}, 2},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 5, 5, 5}}, {{ov::Shape{5, 5, 5, 5, 5}}}}, 2},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 5, 1, 1}}, {{ov::Shape{5, 5, 5, 1, 1}}}}, 3},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 5, 5, 5}}, {{ov::Shape{5, 5, 5, 5, 5}}}}, 3},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 5, 5, 1}}, {{ov::Shape{5, 5, 5, 5, 1}}}}, 4},
+    {ShapesDefenition{{ov::PartialShape{5, 5, 5, 5, 5}}, {{ov::Shape{5, 5, 5, 5, 5}}}}, 4},
+    // Dynamic shapes
+    {ShapesDefenition{{// dynamic shape
+                       {-1, -1}},
+                      {// target static shapes
+                       {{10, 10}},
+                       {{15, 15}},
+                       {{10, 5}}}},
```

Comment:
To test the cache it makes sense to add repeated target shapes so as the cache is triggered. Something like that
```suggestion
                       {{10, 10}},
                       {{15, 15}},
                       {{10, 10}},
                       {{10, 5}}}},
```

Reply:
Yes, I add one repeated tests `{1, 100}` in previous softMax_Optimized tests. But the more the better.